File: Java/com.faculdadevincit.designpatterns.factorypattern/src/TransporteFactory.java
public class TransporteFactory {
    public static ITransporte createTransporte(TransporteEnum transporteEnum) {
        switch (transporteEnum) {
            case AQUATICO:
                return new TransporteAquatico();
            case FERROVIARIO:
                return new TransporteFerroviario();
            case RODOVIARIO:
                return new TransporteRodoviario();
        }

        throw new UnsupportedOperationException();
    }
}

File: CSharp/FaculdadeVincit.DesignPatterns.FactoryPattern/TransporteFactory.cs
namespace FaculdadeVincit.DesignPatterns.FactoryPattern;

public static class TransporteFactory
{
    public static ITransporte CreateTransporte(TransporteEnum transporteEnum)
    {
        switch (transporteEnum)
        {
            case TransporteEnum.AQUATICO:
                return new TransporteAquatico();
            case TransporteEnum.FERROVIARIO:
                return new TransporteFerroviario();
            case TransporteEnum.RODOVIARIO:
                return new TransporteRodoviario();
        }

        throw new NotImplementedException();
    }
}
